pylint/checkers/variables.py

```diff
@@ -695,26 +694,32 @@ def get_next_to_consume(self, node: nodes.Name) -> list[nodes.NodeNG] | None:
         return found_nodes
 
     @staticmethod
-    def _exhaustively_define_name_raise_or_return(
-        name: str, node: nodes.NodeNG
-    ) -> bool:
+    def _inferred_to_define_name_raise_or_return(name: str, node: nodes.NodeNG) -> bool:
         """Return True if there is a collectively exhaustive set of paths under
         this `if_node` that define `name`, raise, or return.
         """
         # Handle try and with
         if isinstance(node, (nodes.TryExcept, nodes.TryFinally)):
             # Allow either a path through try/else/finally OR a path through ALL except handlers
-            return (
-                NamesConsumer._defines_name_raises_or_returns_recursive(name, node)
-                or isinstance(node, nodes.TryExcept)
-                and all(
-                    NamesConsumer._defines_name_raises_or_returns_recursive(
-                        name, handler
-                    )
-                    for handler in node.handlers
+            try_except_node = node
+            if isinstance(node, nodes.TryFinally):
+                try_except_node = next(
+                    (
+                        child
+                        for child in node.get_children()
+                        if isinstance(child, nodes.TryExcept)
```

Comment:
I think this is equivalent.
```suggestion
                        for child in node.nodes_of_class(nodes.TryExcept)
```